### PR TITLE
sync protocol

### PR DESCRIPTION
**Pull request by iznauy**

### CODE REVIEW

**Review by zhuyuqing — CHANGES_REQUESTED** [review]
Please provide an adequate description on the functionality of the PR.

File: sync/src/main/java/cn/edu/tsinghua/iginx/protocol/SyncProtocol.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package cn.edu.tsinghua.iginx.protocol;
+
+import cn.edu.tsinghua.iginx.proposal.ProposalListener;
+import cn.edu.tsinghua.iginx.proposal.SyncProposal;
+import cn.edu.tsinghua.iginx.proposal.SyncVote;
+import cn.edu.tsinghua.iginx.proposal.VoteListener;
+
```
Comment (zhuyuqing) [inline sync/src/main/java/cn/edu/tsinghua/iginx/protocol/SyncProtocol.java:7]: Please add adequate comments and descriptions on the interface.

File: sync/src/main/java/cn/edu/tsinghua/iginx/proposal/ProposalListener.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package cn.edu.tsinghua.iginx.proposal;
+
+public interface ProposalListener {
+
```
Comment (zhuyuqing) [inline sync/src/main/java/cn/edu/tsinghua/iginx/proposal/ProposalListener.java:4]: Please add definition descriptions on the interface.

File: shared/src/main/java/cn/edu/tsinghua/iginx/utils/JsonUtils.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package cn.edu.tsinghua.iginx.utils;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import java.nio.charset.StandardCharsets;
+
```
Comment (zhuyuqing) [inline shared/src/main/java/cn/edu/tsinghua/iginx/utils/JsonUtils.java:7]: Why use a Json parser different from that used in other modules?
Consider changes.

**Review by zhuyuqing — DISMISSED** [review]
Please fix accordingly.

**Review by zhanglingzhe0820 — APPROVED** [review]

**Pull request merged**